### PR TITLE
176895718 sampleset measurements conversion

### PR DESCRIPTION
**Pull request by mstechly**

### CODE REVIEW

**Review by dexter2206 — CHANGES_REQUESTED** [review]
I left minor comments, but am overall unsure about using Measurements->Sampleset conversion with sample sets of `SPIN` vartype. I'd like to clarify the following:

- The tests mention only using `qubo`
- The docstrings mention that arbitrary BQM can be used when converting from measurements
- I thought that our Measurements objects  are supposed to be used with 0-1 variabes (but I might be wrong). If this is the case, then it should be prohibited to convert from measurements to SampleSet with bqm of `SPIN` type.

File: src/python/tests/test_conversion.py [hunk 1]
```diff
@@ -40,3 +44,78 @@ def test_qubo_with_non_binary_fractions():
...
+        (0, 0, 1),
+        (0, 1, 0),
+        (0, 1, 1),
+        (1, 0, 0),
+        (1, 1, 0),
+        (1, 1, 1),
+        (1, 0, 1),
+        (0, 0, 1),
+    ]
+    energies = [0 for i in range(len(bitstrings))]
```
[the remark this answers is not in the document]
Comment (mstechly) [reply src/python/tests/test_conversion.py:62]: Addressed in https://github.com/zapatacomputing/z-quantum-qubo/pull/8/commits/0963b6dcd918297a6dd0d465e1ed9d59fd2deda4

File: src/python/zquantum/qubo/conversions.py [hunk 2]
```diff
@@ -61,3 +65,43 @@ def convert_openfermion_ising_to_qubo(operator: IsingOperator):
...
+    Args:
+        sampleset: SampleSet we want to convert
+    Returns:
+        Measurements object
+
+    """
+    bitstrings = []
+    for sample in sampleset.samples():
+        bitstring = tuple(sample[i] for i in range(len(sample)))
+        bitstrings.append(bitstring)
```
[the remark this answers is not in the document]
Comment (mstechly) [reply src/python/zquantum/qubo/conversions.py]: Addressed in https://github.com/zapatacomputing/z-quantum-qubo/pull/8/commits/0963b6dcd918297a6dd0d465e1ed9d59fd2deda4

File: src/python/zquantum/qubo/conversions.py [hunk 3]
```diff
@@ -61,3 +65,43 @@ def convert_openfermion_ising_to_qubo(operator: IsingOperator):
         linear_terms, quadratic_terms, offset, vartype="SPIN"
     )
     return dimod_ising.change_vartype("BINARY", inplace=False)
+
+
+def convert_sampleset_to_measurements(sampleset: SampleSet) -> Measurements:
+    """
+    Converts dimod SampleSet to zquantum.core Measurements.
+    NOTE: Since Measurements doesn't hold information about the energy of the samples, this conversion is lossy.
```
[the remark this answers is not in the document]
Comment (mstechly) [reply src/python/zquantum/qubo/conversions.py]: Good points, addressed in https://github.com/zapatacomputing/z-quantum-qubo/pull/8/commits/0963b6dcd918297a6dd0d465e1ed9d59fd2deda4

**Review by dexter2206 — APPROVED** [review]
LGTM

**Pull request merged**